addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1914,6 +1914,18 @@ options.registry.SnippetMove = options.Class.extend({
         if ($allOptions.find('we-customizeblock-option').length <= 1) {
             $allOptions.addClass('d-none');
         }
+
+        // OSU TODO : Something put back the buttons in a visible state but I don't know what. Otherwise the code works.
+        let $overlayArea = this.$overlay.find('.o_overlay_move_options');
+        $overlayArea.find('we-button').removeClass('d-none');
+
+        if (!this.$target.prev().length) {
+            $overlayArea.find('we-button').first().addClass('d-none');
+        }
+
+        if (!this.$target.next().length) {
+            $overlayArea.find('we-button').last().addClass('d-none');
+        }
```

Comment:
Use `_computeWidgetVisibility`

Reply:
I don't see any example in the code to help me understand this function.
I should override it to avoid its default behavior ?

Reply:
You are using an editor with a CTRL-SHIFT-F feature? There are many examples of code using the function.....

- Add a `data-name` on the button widgets.
- Override the method to return true or false depending on whether or not the widget linked to the received name should be visible or not.